siliconcompiler/scheduler/taskscheduler.py

```
@@ -424,5 +422,12 @@ def check(self):
         unreached = set(exit_steps).difference(completed_steps)
 
         if unreached:
-            raise SCRuntimeError(
-                f'These final steps could not be reached: {", ".join(sorted(unreached))}')
+            errors = set([f"{step}/{index}" for step, index in self.__runtime_flow.get_nodes()
+                          if NodeStatus.is_error(self.__record.get("status",
+                                                                   step=step, index=index))])
+            if errors:
+                raise SCRuntimeError(
+                    f'Could not run final steps ({", ".join(sorted(unreached))}) due to a errors in: '
+                    f'{", ".join(sorted(errors))}')
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)